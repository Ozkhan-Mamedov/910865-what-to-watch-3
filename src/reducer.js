import {ALL_GENRES, CHANGE_FILTER_BY_GENRE, GET_FILMS_BY_GENRE} from "./constants";
import films from "./mocks/films";
import extend from "./utils";

const initialState = {
  genre: ALL_GENRES,
  films,
};

const ActionType = {
  CHANGE_FILTER_BY_GENRE,
  GET_FILMS_BY_GENRE,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_FILTER_BY_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case GET_FILMS_BY_GENRE:
      return extend(state, {
        films: films.filter((it) => it.genre === state.genre),
      });
  }

  return state;
};

export {reducer, ActionType};
