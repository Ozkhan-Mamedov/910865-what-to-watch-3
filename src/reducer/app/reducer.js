import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType} from "./action-creator";

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_FILTER_BY_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case ActionType.CHANGE_ACTIVE_CARD:
      return extend(state, {
        activeCard: action.payload,
      });

    case ActionType.INCREASE_CARDS_NUMBER:
      return extend(state, {
        cardsRenderNumber: state.cardsRenderNumber + action.payload,
      });

    case ActionType.DECREASE_CARDS_NUMBER:
      return extend(state, {
        cardsRenderNumber: action.payload,
      });

    case ActionType.CHANGE_SERVER_STATUS:
      return extend(state, {
        isServerAvailable: action.payload,
      });
  }

  return state;
};

export {reducer};
