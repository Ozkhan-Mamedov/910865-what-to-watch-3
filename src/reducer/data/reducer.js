import {commentAdapter, filmAdapter} from "../../adapter";
import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType, ActionCreator} from "./action-creator";

const Operation = {
  getFilmList: () => (dispatch, getState, api) => {
    return api.get(`/films`)
      .then((responce) => {
        dispatch(ActionCreator.getFilmList(
            responce.data.map((film) => filmAdapter(film))
        ));
      });
  },
  getCommentsList: (id) => (dispatch, getState, api) => {
    return api.get(`/comments/${id + 1}`)
      .then((responce) => {
        dispatch(ActionCreator.getCommentsList(
            responce.data.map((comment) => commentAdapter(comment))
        ));
      });
  },
  getPromoMovieData: () => (dispatch, getState, api) => {
    return api.get(`/films/promo`)
      .then((responce) => {
        dispatch(ActionCreator.getPromoMovieData(filmAdapter(responce.data)));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.GET_FILM_LIST:
      return extend(state, {
        films: action.payload,
      });

    case ActionType.GET_COMMENTS_LIST:
      return extend(state, {
        filmsComments: action.payload,
      });

    case ActionType.GET_PROMO_MOVIE_DATA:
      return extend(state, {
        promoFilm: action.payload,
      });
  }

  return state;
};

export {reducer, Operation};
