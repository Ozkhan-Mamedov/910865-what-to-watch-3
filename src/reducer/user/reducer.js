import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType, ActionCreator} from "./action-creator";
import {AuthorizationStatus} from "../../constants";

const Operation = {
  checkUserStatus: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.changeAuthorizationStatus(AuthorizationStatus.AUTH));
      });
  },

  login: (authData, handleErrorRequest) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then(() => {
        dispatch(ActionCreator.changeAuthorizationStatus(AuthorizationStatus.AUTH));
      })
      .catch(() => handleErrorRequest());
  }
};

const reducer = (state = initialState, action) => {
  if (action.type === ActionType.CHANGE_AUTHORIZATION_STATUS) {
    return extend(state, {
      authorizationStatus: action.payload,
    });
  }

  return state;
};

export {reducer, Operation};
