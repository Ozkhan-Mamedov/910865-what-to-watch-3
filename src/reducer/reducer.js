import {combineReducers} from "redux";

import {reducer as data} from "./data/reducer";
import {reducer as app} from "./app/reducer";
import NameSpace from "./name-space";

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.APP]: app,
});
