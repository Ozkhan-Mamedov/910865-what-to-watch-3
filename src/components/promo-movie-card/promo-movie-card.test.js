import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {createStore} from "redux";

import PromoMovieCard from "./promo-movie-card";

import reducer from "../../reducer/reducer";

const promoFilmData = {
  name: `Film#1`,
  picture: `picture#1`,
  genre: `genre#1`,
  id: 1,
  releaseDate: `2020`,
  ratingScore: 1,
  ratingsNumber: 2,
  director: `Director#1`,
  starring: [`Actor#1`, `Actor#2`, `Actor#3`, `Actor#4`],
  description: `Test paragraph #1.`,
  preview: `preview#1`,
  runTime: 130,
  previewImage: `preview image#1`,
  videoLink: `video link#1`,
  isFavorite: false,
  backgroundColor: `background color#1`,
  backgroundImage: `background image#1`,
};

it(`PromoMovieCard component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(reducer)}>
          <PromoMovieCard promoFilm={promoFilmData} />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
