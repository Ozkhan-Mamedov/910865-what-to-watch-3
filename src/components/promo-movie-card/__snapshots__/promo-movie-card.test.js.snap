// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoMovieCard component renders correctly 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Film#1"
      src="background image#1"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head movie-card__head"
    style={Object {}}
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Film#1 poster"
          height="327"
          src="preview image#1"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Film#1
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            genre#1
          </span>
          <span
            className="movie-card__year"
          >
            2020
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/0"
            onClick={[Function]}
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
